boards: lilygo: ttgo_lora32: enable SDHC support

lfelten opened the pull request:
enable support for the SDHC controller to use the micro SD card slot

lfelten committed:
boards: lilygo: ttgo_lora32: enable SDHC support

device tree:
enable support for the SDHC controller to use the micro SD card slot

documentation:
- added instructions for SD card and OLED samples
- added links to code samples

defconfig:
added CONFIG_ESP32_USE_UNSUPPORTED_REVISION=y to
ttgo_lora32_esp32_procpu_defconfig

The chip on the board is a ESP32 chip revision 1.
The board will not boot, it displays the following warning at boot:

I (35) boot: chip revision: v1.0
E (38) boot: You are using ESP32 chip revision (1) that is unsupported.
While it may work, it could cause unexpected behavior or issues.
E (50) boot: Proceeding with this ESP32 chip revision is not recommended
unless you fully understand the potential risk and limitations.
E (62) boot: If you choose to continue, please enable the
'CONFIG_ESP32_USE_UNSUPPORTED_REVISION' in your project configuration.
E (73) boot: HW init failed, aborting

In order to prevent a boot loop, CONFIG_ESP32_USE_UNSUPPORTED_REVISION=y
was added to the defconfig.

Signed-off-by: Lothar Felten <lothar.felten@gmail.com>

danieldegrasse reviewed (commented):
Can you add the `sdhc` tag to the list of features this board supports so the SDHC tests run on it? https://github.com/zephyrproject-rtos/zephyr/blob/73a3438b8204a61211b5f78786799a8e6ba77afa/boards/nxp/lpcxpresso55s69/lpcxpresso55s69_lpc55s69_cpu0.yaml#L27

Also, we should update the documentation page for this board to indicate SDHC support

lfelten commented:
Thanks for you feedback!

> Can you add the `sdhc` tag to the list of features this board supports so the SDHC tests run on it?
added

> Also, we should update the documentation page for this board to indicate SDHC support
It mentioned SD before, I updated it to SDHC

sylvioalves reviewed (changes requested):
Now, add ' sdhc'  to the ignore list in `ttgo_lora32_esp32_appcpu.yaml`.

raffarost commented on boards/lilygo/ttgo_lora32/ttgo_lora32-pinctrl.dtsi:
```diff
@@ -21,6 +21,18 @@
 		};
 	};
 
+	spim2_default: spim2_default {
```
this is not related to SDHC, right?

lfelten replied:
Hello Raffael,
no, thankse.  I removed that.

lfelten commented:
> Now, add ' sdhc' to the ignore list in `ttgo_lora32_esp32_appcpu.yaml`.

Added it. 
I also added CONFIG_ESP32_USE_UNSUPPORTED_REVISION=y in the procpu defconfig, which is required since I rebased to main

lfelten commented:
This fails during linking, undefined symbols are ctime_r and asctime_r.

sylvioalves commented:
> This fails during linking, undefined symbols are ctime_r and asctime_r. I'll rebase to main.

Indeed. Fix in #79455.

sylvioalves commented:
@lfelten please rebase.

sylvioalves commented:
@lfelten Is this completed?

sylvioalves commented on boards/lilygo/ttgo_lora32/doc/index.rst:
```diff
@@ -205,7 +205,7 @@ message in the monitor:
    Hello World! ttgo_lora32
 
-LoRa samples
+Code samples
```
@kartben Sounds this file was not updated to latest doc design. Should we do it before v4.0.0?

kartben replied:
sure, we can try :) #81411

lfelten replied:
It didn't build out of the box because the device tree lacked "disk-name = "SD";".
I added it, it now builds, and I tested it with the "fs_sample".
Also rebased on to v4.0.0

kartben commented on boards/lilygo/ttgo_lora32/ttgo_lora32_esp32_procpu_defconfig:
```diff
@@ -8,3 +8,4 @@ CONFIG_UART_CONSOLE=y
 
 CONFIG_GPIO=y
 CONFIG_I2C=y
+CONFIG_ESP32_USE_UNSUPPORTED_REVISION=y
```
Can you add a comment on why this is needed (if this is needed?). This way it's not only clearer to the user but also helps knowing when we might be able to remove it in the future

kartben replied:
At a minimum, makes this a different (and first) commit, similar to https://github.com/zephyrproject-rtos/zephyr/commit/956edd1c3dbf539f1e4b472c61e78668c5e85a16 since my understanding is that it is an unrelated (but needed) change

kartben replied:
fwiw i was not suggesting to add a comment to the git commit (although it's not a bad idea to do it too :) ) but rather right in the defconfig file so that people understand where this is coming from. BTW I am not sure this really belongs to the defconfig as this is not something that should be modifiable, no? Shouldn't it be `select`ed in the Kconfig file instead? cc @sylvioalves

lfelten replied:
It says "unsupported revision" - at least my board uses this version of the chip.
Maybe newer boards (it's still in production) use a newer version.
I'm not sure this setting has any impact when selected for a chip revision > 1

 I added a comment and also trimmed the commit log line length.

sylvioalves replied:
```
config ESP32_USE_UNSUPPORTED_REVISION
	bool "Use unsupported ESP32 revision (Rev 0/1)"
	help
	  ESP32 SoC has multiple revisions, some of which are not supported by the current
	  implementation, as such as REV0 and REV1. In case those revisions are required,
	  set this option to enable support for them. Note that this is not recommended and
	  may lead to unexpected behavior.
```

rev0 and rev1 ESP32 SocS (old versions) have rom bugs that we haven't and will not address. So if the board used such old chip revision, this config must be set.

kartben commented on boards/lilygo/ttgo_lora32/doc/index.rst:
```diff
@@ -220,6 +220,26 @@ To build the LoRa receive sample application using sysbuild use the command:
    :west-args: --sysbuild
    :compact:
 
+The following sample application lists the contents of the SD card:
+
+.. zephyr-app-commands::
+   :tool: west
+   :zephyr-app: samples/subsys/fs/fs_sample
+   :board: ttgo_lora32/esp32/procpu
+   :goals: build
+   :west-args: --sysbuild
+   :compact:
+
+The following sample application displays numbers on the OLED display:
+
+.. zephyr-app-commands::
+   :tool: west
+   :zephyr-app: samples/subsys/display/cfb
+   :board: ttgo_lora32/esp32/procpu
+   :goals: build
+   :west-args: --sysbuild
+   :compact:
```
Not blocking but I'd rather you link to the samples instead. I understand that this is useful to give ready to copy-paste build instructions but it's also content that might be left behind should something change in the way these samples should be built (arguably unlikely, but still)
I would simply say something like "You may also run :zephyr:code-sample:`XXX` to blah blah or ... to ...".

lfelten replied:
Hi @kartben 
Yes, I can change them to be links.
Since I usually compare with documentation files from other boards, is there a reference documentation you'd recommend? Especially how to make the links.
The copy-paste build instructions did actually change not so long ago, and I had to update them. So, removing those would be good.

lfelten replied:
Should I add links like this?

* :zephyr:code-sample:`lora-send`
* :zephyr:code-sample:`lora-receive`
* :zephyr:code-sample:`fs-fs-sample`
* :zephyr:code-sample:`display-cfb`

(how do subdirectories/catgories get handled?)

kartben reviewed (changes requested):
A nit and an actual request for change. Thanks for rebasing, too!
